src/Reflector/DefaultReflector.php

```diff
@@ -56,6 +57,23 @@ public function reflectAllClasses(): iterable
         return $allClasses;
     }
 
+    /**
+     * Iterate classes available in the scope specified by the SourceLocator.
+     *
+     * @return Generator<ReflectionClass>
+     */
+    public function iterateClasses(?SourceFilter $filter = null): Generator
+    {
+        /** @var Generator<ReflectionClass> $allClasses */
+        $allClasses = $this->sourceLocator->iterateIdentifiersByType(
+            $this,
+            new IdentifierType(IdentifierType::IDENTIFIER_CLASS),
+            $filter,
+        );
+
+        yield from $allClasses;
+    }
```

Comment:
I don't think `SourceFilter` should be given at call-time: it should be an implementation detail of the source locator only, IMO.